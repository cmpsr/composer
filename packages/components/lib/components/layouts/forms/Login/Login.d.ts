import React from "react";
import { Asset as AssetType } from "types";
declare type Props = {
    className?: string;
    asset?: AssetType;
    title?: string;
    remember?: string;
    forgot?: {
        title?: string;
        url?: string;
        route?: {
            slug: string;
        };
    };
    signIn?: {
        icon?: React.ReactNode;
        title?: string;
    };
    onSubmit?: (credentials: {
        name: string;
        password?: string;
        remember?: boolean;
    }) => void;
};
export declare const Login: ({ className, asset, title, remember, forgot, signIn, onSubmit, }: Props) => JSX.Element;
export {};
//# sourceMappingURL=Login.d.ts.map