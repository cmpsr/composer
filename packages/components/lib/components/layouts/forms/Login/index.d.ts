export * from './Login';
//# sourceMappingURL=index.d.ts.map