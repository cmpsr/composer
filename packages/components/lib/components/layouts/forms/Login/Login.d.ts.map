{"version":3,"file":"Login.d.ts","sourceRoot":"","sources":["../../../../../src/components/layouts/forms/Login/Login.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAc1B,OAAO,EAAE,KAAK,IAAI,SAAS,EAAE,MAAM,OAAO,CAAC;AAE3C,aAAK,KAAK,GAAG;IACX,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,KAAK,CAAC,EAAE,SAAS,CAAC;IAClB,KAAK,CAAC,EAAE,MAAM,CAAC;IACf,QAAQ,CAAC,EAAE,MAAM,CAAC;IAClB,MAAM,CAAC,EAAE;QACP,KAAK,CAAC,EAAE,MAAM,CAAC;QACf,GAAG,CAAC,EAAE,MAAM,CAAC;QACb,KAAK,CAAC,EAAE;YAAE,IAAI,EAAE,MAAM,CAAA;SAAE,CAAC;KAC1B,CAAC;IACF,MAAM,CAAC,EAAE;QACP,IAAI,CAAC,EAAE,KAAK,CAAC,SAAS,CAAC;QACvB,KAAK,CAAC,EAAE,MAAM,CAAC;KAChB,CAAC;IACF,QAAQ,CAAC,EAAE,CAAC,WAAW,EAAE;QACvB,IAAI,EAAE,MAAM,CAAC;QACb,QAAQ,CAAC,EAAE,MAAM,CAAC;QAClB,QAAQ,CAAC,EAAE,OAAO,CAAC;KACpB,KAAK,IAAI,CAAC;CACZ,CAAC;AAEF,eAAO,MAAM,KAAK,qEAQf,KAAK,gBA2GP,CAAC"}