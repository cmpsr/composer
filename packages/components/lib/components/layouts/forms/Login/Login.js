"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Login = void 0;

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _primitives = require("../../../primitives");

var _TextField = require("../../../primitives/form/TextField");

var _Link = require("../../../primitives/Link");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const Login = ({
  className,
  asset,
  title,
  remember,
  forgot,
  signIn = {
    title: "Sign in"
  },
  onSubmit
}) => {
  const [username, setUsername] = (0, _react.useState)();
  const [password, setPassword] = (0, _react.useState)();
  const [rememberMe, setRememberMe] = (0, _react.useState)();

  const onChangeUsername = event => setUsername(event.currentTarget.value);

  const onChangePassword = event => setPassword(event.currentTarget.value);

  const onChangeRememberMe = event => setRememberMe(event.currentTarget.checked);

  const onFormSubmit = e => {
    e.preventDefault();
    onSubmit({
      name: username,
      password,
      remember: rememberMe
    });
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)("h-screen flex items-center justify-center bg-gray-50 px-4 lg:px-8", className)
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "max-w-md w-full"
  }, /*#__PURE__*/_react.default.createElement("div", null, asset && /*#__PURE__*/_react.default.createElement(_primitives.Asset, {
    className: "mx-auto h-20 w-auto",
    asset: asset
  }), /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "h2",
    type: _primitives.TypographyTypes.Headline4,
    className: "text-center mt-8"
  }, title)), /*#__PURE__*/_react.default.createElement("form", {
    className: "mt-8",
    onSubmit: onFormSubmit
  }, /*#__PURE__*/_react.default.createElement("input", {
    type: "hidden",
    name: "remember",
    value: "true"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "rounded-md shadow-sm"
  }, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_TextField.TextField, {
    ariaLabel: "Name",
    name: "name",
    placeholder: "Name",
    required: true,
    onChange: onChangeUsername
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-2"
  }, /*#__PURE__*/_react.default.createElement(_TextField.TextField, {
    ariaLabel: "Password",
    name: "password",
    placeholder: "Password",
    type: "password",
    onChange: onChangePassword
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-6 flex items-center justify-between"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex items-center"
  }, remember && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("input", {
    id: "remember_me",
    type: "checkbox",
    className: "form-checkbox h-4 w-4 fill-primary-100 transition duration-150 ease-in-out",
    onChange: onChangeRememberMe
  }), /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: "remember_me",
    className: "ml-2 block text-sm leading-5 text-gray-900"
  }, remember))), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-sm leading-5"
  }, (forgot === null || forgot === void 0 ? void 0 : forgot.title) && /*#__PURE__*/_react.default.createElement(_Link.Link, {
    className: "font-medium fill-primary-100 focus:outline-none focus:underline transition ease-in-out duration-150",
    url: forgot.url,
    route: forgot.route
  }, forgot.title))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-6"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Button, {
    htmlType: "submit",
    type: _primitives.ButtonTypes.Primary,
    className: "w-full relative"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute left-0 px-4"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Icon, {
    className: "fill-current",
    width: 24,
    height: 24
  }, signIn === null || signIn === void 0 ? void 0 : signIn.icon)), signIn === null || signIn === void 0 ? void 0 : signIn.title)))));
};

exports.Login = Login;