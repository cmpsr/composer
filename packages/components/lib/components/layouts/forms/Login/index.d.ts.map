{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../../src/components/layouts/forms/Login/index.ts"],"names":[],"mappings":"AAAA,cAAc,SAAS,CAAC"}