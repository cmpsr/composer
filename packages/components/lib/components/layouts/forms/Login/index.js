"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Login = require("./Login");

Object.keys(_Login).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _Login[key];
    }
  });
});