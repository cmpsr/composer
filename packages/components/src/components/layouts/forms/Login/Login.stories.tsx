import React from 'react';
import { action } from '@storybook/addon-actions';
import { lock as Lock } from 'components/primitives/Icon/icons/action';

import { Login } from './Login';

export default {
  title: 'Composer/Forms/Login',
  component: Login,
};

export const withTitle = () => <Login title="Sign in to your account" />;

withTitle.story = {
  name: 'With Title',
};

export const withAsset = () => (
  <Login
    asset={{
      contentType: 'img',
      title: 'Image Title',
      url:
        'https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg',
    }}
  />
);

withAsset.story = {
  name: 'With Asset',
};

export const withSubmit = () => (
  <Login onSubmit={action('on submit form action')} />
);

withSubmit.story = {
  name: 'With Submit Action',
};

withSubmit.story = {
  name: 'With Submit Action',
};

export const withRemember = () => <Login remember="Remember me" />;

withRemember.story = {
  name: 'With Remember me',
};

export const withForgotPassword = () => (
  <Login forgot={{ title: 'Forgot your password?' }} />
);

withForgotPassword.story = {
  name: 'With Forgot Password',
};

export const withSignInTitleAndIcon = () => {
  const icon = <Lock type="filled" />;
  return <Login signIn={{ title: 'Sign in title', icon }} />;
};

withSignInTitleAndIcon.story = {
  name: 'With Sign in title & Icon',
};

export const withAll = () => {
  const icon = <Lock type="filled" />;
  return (
    <Login
      asset={{
        contentType: 'img',
        title: 'Image Title',
        url:
          'https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg',
      }}
      title="Sign in to your account"
      remember="Remember me"
      forgot={{ title: 'Forgot your password?' }}
      signIn={{ title: 'Sign in title', icon }}
    />
  );
};

withAll.story = {
  name: 'With all properties set',
};
