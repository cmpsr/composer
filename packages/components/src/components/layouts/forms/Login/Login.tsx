import React from "react";
import cn from "classnames";

import {
  Button,
  ButtonTypes,
  Typography,
  TypographyTypes,
  Icon,
  Asset,
} from "components/primitives";
import { useState, ChangeEvent } from "react";
import { TextField } from "components/primitives/form/TextField";
import { Link } from "components/primitives/Link";
import { Asset as AssetType } from "types";

type Props = {
  className?: string;
  asset?: AssetType;
  title?: string;
  remember?: string;
  forgot?: {
    title?: string;
    url?: string;
    route?: { slug: string };
  };
  signIn?: {
    icon?: React.ReactNode;
    title?: string;
  };
  onSubmit?: (credentials: {
    name: string;
    password?: string;
    remember?: boolean;
  }) => void;
};

export const Login = ({
  className,
  asset,
  title,
  remember,
  forgot,
  signIn = { title: "Sign in" },
  onSubmit,
}: Props) => {
  const [username, setUsername] = useState<string>();
  const [password, setPassword] = useState<string>();
  const [rememberMe, setRememberMe] = useState<boolean>();

  const onChangeUsername = (event: ChangeEvent<HTMLInputElement>) =>
    setUsername(event.currentTarget.value);
  const onChangePassword = (event: ChangeEvent<HTMLInputElement>) =>
    setPassword(event.currentTarget.value);
  const onChangeRememberMe = (event: ChangeEvent<HTMLInputElement>) =>
    setRememberMe(event.currentTarget.checked);
  const onFormSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    onSubmit({ name: username, password, remember: rememberMe });
  };

  return (
    <div
      className={cn(
        "h-screen flex items-center justify-center bg-gray-50 px-4 lg:px-8",
        className
      )}
    >
      <div className="max-w-md w-full">
        <div>
          {asset && <Asset className="mx-auto h-20 w-auto" asset={asset} />}
          <Typography
            tag="h2"
            type={TypographyTypes.Headline4}
            className="text-center mt-8"
          >
            {title}
          </Typography>
        </div>
        <form className="mt-8" onSubmit={onFormSubmit}>
          <input type="hidden" name="remember" value="true" />
          <div className="rounded-md shadow-sm">
            <div>
              <TextField
                ariaLabel="Name"
                name="name"
                placeholder="Name"
                required
                onChange={onChangeUsername}
              />
            </div>
            <div className="mt-2">
              <TextField
                ariaLabel="Password"
                name="password"
                placeholder="Password"
                type="password"
                onChange={onChangePassword}
              />
            </div>
          </div>
          <div className="mt-6 flex items-center justify-between">
            <div className="flex items-center">
              {remember && (
                <>
                  <input
                    id="remember_me"
                    type="checkbox"
                    className="form-checkbox h-4 w-4 fill-primary-100 transition duration-150 ease-in-out"
                    onChange={onChangeRememberMe}
                  />
                  <label
                    htmlFor="remember_me"
                    className="ml-2 block text-sm leading-5 text-gray-900"
                  >
                    {remember}
                  </label>
                </>
              )}
            </div>

            <div className="text-sm leading-5">
              {forgot?.title && (
                <Link
                  className="font-medium fill-primary-100 focus:outline-none focus:underline transition ease-in-out duration-150"
                  url={forgot.url}
                  route={forgot.route}
                >
                  {forgot.title}
                </Link>
              )}
            </div>
          </div>

          <div className="mt-6">
            <Button
              htmlType="submit"
              type={ButtonTypes.Primary}
              className="w-full relative"
            >
              <span className="absolute left-0 px-4">
                <Icon className="fill-current" width={24} height={24}>
                  {signIn?.icon}
                </Icon>
              </span>
              {signIn?.title}
            </Button>
          </div>
        </form>
      </div>
    </div>
  );
};
