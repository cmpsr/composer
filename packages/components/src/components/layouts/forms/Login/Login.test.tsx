import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Login } from './Login';

describe('Login', () => {
  it('should match snapshot', () => {
    const login = renderer.create(
      <Login
        remember="Remember me"
        forgot={{ title: 'Forgot your password?' }}
      />,
    );
    const tree = login.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render asset', () => {
    const title = 'img alt';
    const { getByAltText } = render(
      <Login asset={{ contentType: 'img', title, url: '' }} />,
    );
    const asset = getByAltText(title);
    expect(asset).toBeInTheDocument();
  });
  it('should render title', () => {
    const title = 'Sign in to your account';
    const { getByText } = render(<Login title={title} />);
    const label = getByText(title);
    expect(label).toBeInTheDocument();
  });
  it('should render user name input field', () => {
    const { getByPlaceholderText } = render(<Login />);
    const nameInput = getByPlaceholderText('Name');
    expect(nameInput).toBeInTheDocument();
  });
  it('should render user password input field', () => {
    const { getByPlaceholderText } = render(<Login />);
    const nameInput = getByPlaceholderText('Password');
    expect(nameInput).toBeInTheDocument();
  });
  it('should render remember me checkbox', () => {
    const { container, getByText } = render(<Login remember="Remember me" />);
    const checkbox = container.querySelector('#remember_me');
    expect(checkbox).toBeInTheDocument();
    const rememberLabel = getByText('Remember me');
    expect(rememberLabel).toBeInTheDocument();
  });
  it('should render forgot password text', () => {
    const title = 'Forgot your password?';
    const { getByText } = render(<Login forgot={{ title }} />);
    const forgotLabel = getByText(title);
    expect(forgotLabel).toBeInTheDocument();
  });
  it('should render sig in title', () => {
    const signInTitle = 'Sign in';
    const { getByText } = render(<Login signIn={{ title: signInTitle }} />);
    const signInButton = getByText(signInTitle);
    expect(signInButton).toBeInTheDocument();
  });
  it('should call onSubmit with credentials', () => {
    const onSubmit = jest.fn();
    const userName = 'user name';
    const password = 'password';
    const { getByPlaceholderText, getByText } = render(
      <Login onSubmit={onSubmit} />,
    );
    const nameInput = getByPlaceholderText('Name');
    const passwordInput = getByPlaceholderText('Password');
    const button = getByText('Sign in');
    fireEvent.change(nameInput, { target: { value: userName } });
    fireEvent.change(passwordInput, { target: { value: password } });
    fireEvent.click(button);
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenCalledWith({
      name: userName,
      password,
      remember: undefined,
    });
  });
  it('should call onSubmit with credentials including remember me', () => {
    const onSubmit = jest.fn();
    const userName = 'user name';
    const password = 'password';
    const { container, getByPlaceholderText, getByText } = render(
      <Login remember="Remember me" onSubmit={onSubmit} />,
    );
    const nameInput = getByPlaceholderText('Name');
    const passwordInput = getByPlaceholderText('Password');
    const button = getByText('Sign in');
    const checkbox = container.querySelector('#remember_me');
    fireEvent.change(nameInput, { target: { value: userName } });
    fireEvent.change(passwordInput, { target: { value: password } });
    fireEvent.click(checkbox);
    fireEvent.click(button);
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenCalledWith({
      name: userName,
      password,
      remember: true,
    });
  });
});
