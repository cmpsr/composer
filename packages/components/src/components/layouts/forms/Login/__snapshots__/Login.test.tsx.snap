// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<div
  className="h-screen flex items-center justify-center bg-gray-50 px-4 lg:px-8"
>
  <div
    className="max-w-md w-full"
  >
    <div>
      <h2
        className="typo-headline-4 text-center mt-8"
        data-testid="typography"
      />
    </div>
    <form
      className="mt-8"
      onSubmit={[Function]}
    >
      <input
        name="remember"
        type="hidden"
        value="true"
      />
      <div
        className="rounded-md shadow-sm"
      >
        <div>
          <input
            aria-label="Name"
            className="form-input border-outline-forms-filled focus:outline-none placeholder-outline-forms-filled focus:border-outline-forms-focus rounded-input-semi-rounded"
            disabled={false}
            name="name"
            onChange={[Function]}
            placeholder="Name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="mt-2"
        >
          <input
            aria-label="Password"
            className="form-input border-outline-forms-filled focus:outline-none placeholder-outline-forms-filled focus:border-outline-forms-focus rounded-input-semi-rounded"
            disabled={false}
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <div
        className="mt-6 flex items-center justify-between"
      >
        <div
          className="flex items-center"
        >
          <input
            className="form-checkbox h-4 w-4 transition duration-150 ease-in-out"
            id="remember_me"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="ml-2 block text-sm leading-5 text-gray-900"
            htmlFor="remember_me"
          >
            Remember me
          </label>
        </div>
        <div
          className="text-sm leading-5"
        >
          <a
            className="flex-grow-0 text-btn font-medium focus:outline-none focus:underline transition ease-in-out duration-150"
            data-testid="link"
          >
            Forgot your password?
          </a>
        </div>
      </div>
      <div
        className="mt-6"
      >
        <button
          className="w-full relative flex focus:outline-none primary-btn disabled:bg-states-contained-disabled disabled:pointer-events-none active:bg-states-contained-pressed hover:bg-states-contained-hover"
          data-testid="button"
          type="submit"
        >
          <span
            className="absolute left-0 px-4"
          >
            <svg
              className="fill-current"
              data-testid="icon"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
            />
          </span>
          Sign in
        </button>
      </div>
    </form>
  </div>
</div>
`;
